mypy/checker.py

```diff
@@ -1476,6 +1476,10 @@ def check_override(self, override: FunctionLike, original: FunctionLike,
                 # are erased, then it is definitely an incompatibility.
 
                 override_ids = override.type_var_ids()
+                if isinstance(override.definition, FuncDef):
+                    type_name = override.definition.info.name()
+                else:
+                    type_name = "Foo"
```

Comment:
I think a better alternative is to not show the note at all if `__eq__` is not a function definition. So you can just set `type_name` to `None` here, and add a check in `argument_incompatible_with_supertype()` that will not show the note if the name is `None`.